app/resolve-guid/route.ts

```diff
@@ -75,7 +75,14 @@ export default class ResolveGuid extends Route {
                 throw new Error(`Unknown GUID referentType: ${guid.referentType}`);
             }
 
-            expanded = this.generateURL(this.routeMap[guid.referentType], params.guid);
+            if (guid.referentType === 'preprint') {
+                const preprint = await this.store.findRecord('preprint', params.guid);
+                const providerId = preprint.belongsTo('provider').id();
+                expanded = this.generateURL(this.routeMap['preprint'], providerId, params.guid);
```

Comment:
Do we need to check if `providerId === config.defaultProvider`?

Reply:
Talked to Eric and we'd stop special-casing for default osf provider.